src/core/frames/frame_renderer.ts

```diff
@@ -8,15 +8,30 @@ export class FrameRenderer extends Renderer<FrameElement> {
   }
 
   async render() {
-    await nextAnimationFrame()
-    this.preservingPermanentElements(() => {
-      this.loadFrameElement()
+    return new Promise<void>(resolve => {
```

Comment:
I have to wrap it in Promise because overwise the [`View#renderSnapshot`](https://github.com/kirillplatonov/turbo/blob/pausable-render/src/core/view.ts#L95) would resolve at the first await `await nextAnimationFrame()` and won't ever reach `this.loadFrameElement()`